maps_dashboards/public/components/map_top_nav/get_top_nav_config.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { i18n } from '@osd/i18n';
+import { TopNavMenuData } from '../../../../../src/plugins/navigation/public';
+import {
+  OnSaveProps,
+  SavedObjectSaveModalOrigin,
+  showSaveModal,
+} from '../../../../../src/plugins/saved_objects/public';
+import { useOpenSearchDashboards } from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import { MapServices } from '../../types';
+
+export const getTopNavConfig = () => {
+  const {
+    services: {
+      notifications: { toasts },
+      i18n: { Context: I18nContext },
+      savedObjects: { client: savedObjectsClient },
+    },
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+  } = useOpenSearchDashboards<MapServices>();
+  const topNavConfig: TopNavMenuData[] = [
+    {
+      iconType: 'save',
+      emphasize: true,
+      id: 'save',
+      label: i18n.translate('maps.topNav.saveMapButtonLabel', {
+        defaultMessage: `Save`,
+      }),
+      run: (_anchorElement) => {
+        const onModalSave = async (onSaveProps: OnSaveProps) => {
+          const savedMap = await savedObjectsClient.create('map', {
```

Comment:
I'm not totally sure of the API here but does this allow you to edit an existing visualization? I remember seeing that we used the save API and not create. But don't take my word on that

Reply:
Good catch, I plan to implement more logic at this part in next PR, so that it can used for new and saved object.